emailage/client.py

```diff
@@ -170,28 +210,60 @@ def request(self, endpoint, **params):
             u'user20180830001%40domain20180830001.com'
         """
         url = self.domain + '/emailagevalidator' + endpoint + '/'
-
-        params = dict(
+        api_params = dict(
             format='json',
             **params
         )
-        params = signature.add_oauth_entries_to_fields_dict(self.secret, params)
-        params['oauth_signature'] = signature.create('GET', url, params, self.hmac_key)
-        params_qs = _url_encode_dict(params)
 
         request_params = {}
-        if self.timeout is not None:
+        if self.timeout is not None and 'timeout':
             request_params['timeout'] = self.timeout
 
-        res = self.session.get(url, params=params_qs, **request_params)
-      
+        if self.http_method == HttpMethods.GET:
+            response = self._perform_get_request(url, api_params, request_params)
+        else:
+            response = self._perform_post_request(url, api_params, request_params)
+
+        if not response:
+            raise ValueError('No response received for request')
+
         # Explicit encoding is necessary because the API returns a Byte Order Mark at the beginning of the contents
-        json_data = res.content.decode(encoding='utf_8_sig')
+        json_data = response.content.decode(encoding='utf_8_sig')
         return json.loads(json_data)
 
+    def _perform_get_request(self, url, api_params, request_params=None):
+
+        api_params = signature.add_oauth_entries_to_fields_dict(self.secret, api_params)
+        api_params['oauth_signature'] = signature.create(HttpMethods.GET, url, api_params, self.hmac_key)
+
+        params_qs = _url_encode_dict(api_params)
+        request_params = request_params or {}
+
+        res = self.session.get(url, params=params_qs, **request_params)
+        return res
+
+    def _perform_post_request(self, url, api_params, request_params=None):
+        signature_fields = dict(format='json')
+
+        signature_fields = signature.add_oauth_entries_to_fields_dict(self.secret, signature_fields)
+
+        signature_fields['oauth_signature'] = signature.create(HttpMethods.POST, url, signature_fields, self.hmac_key)
+        url = url + '?' + _url_encode_dict(signature_fields)
+
+        payload = bytes(self._assemble_quoted_pairs(api_params), encoding='utf_8')
```

Comment:
this causes incompatibility with python 2.7, which can be easily averted by using `encode('utf-8')` instead:

```
payload = self._assemble_quoted_pairs(api_params).encode('utf_8')
```